torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -153,8 +153,10 @@ void ProcessGroupAgent::join() {
   //    effort to fix this problem).
   sync();
   int dst = (pg_->getRank() + 1) % pg_->getSize();
+  auto future = std::make_shared<FutureMessage>();
   enqueueSend(
-      SendWork(workerIds_[dst], Message({}, {}, MessageType::SHUTDOWN)));
+      SendWork(workerIds_[dst], Message({}, {}, MessageType::SHUTDOWN)),
+      future);
```

Comment:
Ahh I see, this future is not added into the `futures_` map. Shall add this to the map as well and use a similar signature for `enqueueSend` and `enqueueRecv` ?